src/engine/specification.md

```diff
@@ -223,43 +255,35 @@ Each of these stages is optional. Exact behavior of a client software during the
 #### Response
 
 * result: `object`
-    - `status`: `enum` - `"VALID" | "INVALID" | "SYNCING" | "ACCEPTED"`
-    - `latestValidHash`: `DATA|null`, 32 Bytes - the hash of the most recent *valid* block in the branch defined by payload and its ancestors
-    - `validationError`: `String|null` - a message providing additional details on the validation error if the payload is deemed `INVALID`
-    - `payloadId`: `DATA|null`, 8 Bytes - identifier of the payload build process or `null`
-* error: code and message set in case an exception happens while validating payload, updating the forkchoice or initiating the payload build process.
+  - `payloadStatus`: [`PayloadStatusV1`](#PayloadStatusV1); values of the `status` field in the context of this method are restricted to the following subset:
+    * `"VALID"`
+    * `"INVALID"`
+    * `"SYNCING"`
+  - `payloadId`: `DATA|null`, 8 Bytes - identifier of the payload build process or `null`
+* error: code and message set in case an exception happens while the validating payload, updating the forkchoice or initiating the payload build process.
 
 #### Specification
 
-1. Client software **SHOULD** initiate the sync process if `forkchoiceState.headBlockHash` references an unknown payload or a payload that can't be validated because requisite data is missing. The sync process is specified in the [Sync process](#sync-process) section. Additionally:
-    * Client software **MUST** return `{status: SYNCING, latestValidHash: null, validationError: null, payloadId: null}` if the sync process has been initiated.
+1. Client software **MAY** initiate the sync process if `forkchoiceState.headBlockHash` references an unknown payload or a payload that can't be validated because requisite data is missing. The sync process is specified in the [Sync](#sync) section.
 
-1. Client software **MAY** skip an update of the forkchoice state and **MAY NOT** begin a payload build process if the payload identified by `forkchoiceState.headBlockHash` hasn't been validated yet. Additionally:
-    * Client software **MUST** return `{status: ACCEPTED, latestValidHash: null, validationError: null, payloadId: null}` in this case.
+1. Client software **MAY** skip an update of the forkchoice state and **MUST NOT** begin a payload build process if `forkchoiceState.headBlockHash` references an ancestor of the head of the canonical chain.
```

Comment:
Should this be any non-leaf value? -- That is you MAY skip it if it isn't a leaf of the block tree

Otherwise you can abuse the rewind (turn this MAY into a MUST) by fcu to a non-canonical branch and then fcu to a non-leaf block in what was previously the canonical chain